examples/redis-cluster-stateful-mig/consul/consul-cluster/consul-cluster/main.tf

```diff
@@ -0,0 +1,257 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ---------------------------------------------------------------------------------------------------------------------
+# THESE TEMPLATES REQUIRE TERRAFORM VERSION 0.12.0 AND ABOVE
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  # The modules has been updated with 0.12 syntax, which means the example is no longer
+  # compatible with any versions below 0.12.
+  required_version = ">= 0.12"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE A REGIONAL MANAGED INSTANCE GROUP TO RUN THE CONSUL SERVERS
+# ---------------------------------------------------------------------------------------------------------------------
+
+# Create the Regional Managed Instance Group where Consul Server will live.
+resource "google_compute_region_instance_group_manager" "consul_server" {
+  project = var.gcp_project_id
+  name    = "${var.cluster_name}-ig"
+
+  base_instance_name = var.cluster_name
+  region             = var.gcp_region
+
+  version {
+    instance_template = google_compute_instance_template.consul_server.self_link
+  }
+
+  # Consul Server is a stateful cluster, so the update strategy used to roll out a new GCE Instance Template must be
+  # a rolling update.
+  update_policy {
+    type                         = var.instance_group_update_policy_type
+    instance_redistribution_type = var.instance_group_update_policy_redistribution_type
+    minimal_action               = var.instance_group_update_policy_minimal_action
+    max_surge_fixed              = var.instance_group_update_policy_max_surge_fixed
+    max_surge_percent            = var.instance_group_update_policy_max_surge_percent
+    max_unavailable_fixed        = var.instance_group_update_policy_max_unavailable_fixed
+    max_unavailable_percent      = var.instance_group_update_policy_max_unavailable_percent
+    # min_ready_sec                = var.instance_group_update_policy_min_ready_sec
+  }
+
+  target_pools = var.instance_group_target_pools
+  target_size  = var.cluster_size
+
+  depends_on = [
+    google_compute_instance_template.consul_server
+  ]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# Create the Instance Template that will be used to populate the Managed Instance Group.
+resource "google_compute_instance_template" "consul_server" {
```

Comment:
dnsmasq seems to respond only when listen-address contains the forwarding rule ip?   

listen-address=10.128.0.32

i couldn't get it to work otherwise.